Aqovia.Utilities.SagaMachine.Tests/StateMachineTests.cs

```diff
@@ -92,33 +91,23 @@ public StateMachineTests()
         [Fact(DisplayName = "Should not allow InitialiseState without Id")]
         public void SagaMachineShouldHaveInitialiseStateId()
         {
-            //Arrange
-            Guid thisId = Guid.NewGuid();
-
             _sagaMachine
                 .WithMessage<HelloMessage>((proccess, msg) => proccess
-                    .InitialiseState((hello) => new TestState
+                    .InitialiseState(hello => new TestState
                     {
                         SomeStateVariable = hello.Message
                     })
                     .Execute()
                 );
 
-            //Act
-
-            //Assert
-            Action shouldThrow = () =>
+            Func<Task> act = async () =>
             {
-
-                _sagaMachine.Handle(new HelloMessage
+                await _sagaMachine.Handle(new HelloMessage
                 {
                     Message = "Initialised"
-                }).Wait();
+                }).ConfigureAwait(false);
             };
-            shouldThrow.ShouldThrow<SagaException>();
-
-
-
+            act.ShouldThrow<SagaException>();
```

Comment:
`act.ShouldThrow<SagaException>()` is not asyc aware, so it uses `Wait()`, and that's the only reason you need `ConfigureAwait(false)`.
Why not use the newly introduced xUnit 2.0's `Assert.ThrowsAsync` and have it done [the recommended way](https://blog.ciber.no/2014/05/19/using-task-configureawaitfalse-to-prevent-deadlocks-in-async-code/) with just await on it and the test being `async`?

It seems that FluentAssertions sadly didn't catch up yet with the `async` keyword, so if you want a nice `act.ShouldThrowAsync<SagaException>()` extension for the blessed way, you need to roll your own:

```
    public static class FluentAssertionMissingFeaturesExtensions
    {
        public static async Task ThrowsAsync<T>(this Func<Task> testCode) where T : Exception
        {
            await Assert.ThrowsAsync<T>(testCode);
        }
    }
```